lib/dao.js

```diff
@@ -623,9 +626,13 @@ DataAccessObject.upsert = function(data, options, cb) {
                 options: options,
               };
 
-              Model.notifyObserversOf('after save', context, function(err) {
-                cb(err, obj);
-              });
+              if (options.notify !== false) {
```

Comment:
That is also not valid because of options.notify being null by default. 

Unless you explicitly say you don't want to be notified. You're getting notified.

```
Model.create(model, {notify: false});
```